src/main/java/org/folio/circulation/domain/MoveRequestService.java

```diff
@@ -38,14 +38,19 @@ public MoveRequestService(RequestRepository requestRepository, RequestPolicyRepo
         .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::findDestinationItem))
         .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getDestinationRequestQueue))
         .thenApply(r -> r.map(this::pagedRequestIfDestinationItemAvailable))
-        .thenCompose(r -> r.after(this::updateRequest))
-        .thenCompose(r -> r.after(updateRequestQueue::onMovedTo))
+        .thenCompose(r -> r.after(this::validateUpdateRequest))
         .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::findSourceItem))
         .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getSourceRequestQueue))
         .thenCompose(r -> r.after(updateRequestQueue::onMovedFrom))
         .thenCompose(r -> r.after(updateUponRequest.updateItem::onRequestQueueChanged))
         .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::findDestinationItem))
-        .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getDestinationRequestQueue));
+        .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getDestinationRequestQueue))
```

Comment:
Why does the destination request queue need fetching twice (it also appears on line 39 above)?

Reply:
Before any validation is performed it fetches the destination item and request queue to perform meaningful validation of what is desired. After success, it fetches the source item and request queue to update them. Then it requires fetching the destination item and request queue to update them and returned the request with its intended item and request queue.

Reply:
Oh, I think I follow, the `request queue` in the related records could be either the source or destination queue, and so the system does not have access to both at once.

Reply:
Correct. Could cache the subsequent items and request queues and toggle between them.